### PR TITLE
Add missing plate and well schemas for 0.1, 0.2, 0.3

### PR DESCRIPTION
**Pull request by will-moore**
As discussed in https://github.com/ome/ome-ngff-validator/pull/3, we can't validate HCS data for v 0.1, 0.2, 0.3 because of missing schemas.

This adds plate schema for 0.2 & 0.3 (copying over the 0.1 schema and updating version)
and adds well schema for 0.1, 0.2, 0.3 (copying the 0.4 schema). This only requires 'path' attribute which is correct for all versions.

cc @sbesson

### CODE REVIEW

File: 0.2/schemas/well.schema [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://ngff.openmicroscopy.org/0.4/schemas/well.schema",
```
Comment (sbesson) [inline 0.2/schemas/well.schema:3]:
```suggestion
  "$id": "https://ngff.openmicroscopy.org/0.2/schemas/well.schema",
```

File: 0.3/schemas/plate.schema [hunk 2]
```diff
@@ -0,0 +1,112 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "http://localhost:8000/plate.schema",
```
Comment (sbesson) [inline 0.3/schemas/plate.schema:3]:
```suggestion
  "$id": "https://ngff.openmicroscopy.org/0.3/schemas/plate.schema"
```

File: 0.3/schemas/well.schema [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://ngff.openmicroscopy.org/0.4/schemas/well.schema",
```
Comment (sbesson) [inline 0.3/schemas/well.schema:3]:
```suggestion
  "$id": "https://ngff.openmicroscopy.org/0.3/schemas/well.schema",
```

File: 0.2/schemas/plate.schema [hunk 4]
```diff
@@ -0,0 +1,112 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "http://localhost:8000/plate.schema",
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:3]:
```suggestion
  "$id": "https://ngff.openmicroscopy.org/0.2/schemas/plate.schema",
```

File: 0.2/schemas/plate.schema [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  "title": "NGFF Plate",
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:4]:
```suggestion
  "title": "OME-NGFF plate schema",
```

File: 0.2/schemas/plate.schema [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+          "description": "Maximum number of fields per view across all wells."
+        },
+        "acquisitions": {
+          "description": "Rows of the Plate grid",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "id": {
+                "type": "number"
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:81]: Specified as `integer` in 0.4/latest schemas

File: 0.2/schemas/plate.schema [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+              },
+              "maximumfieldcount": {
+                "type": "number"
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:84]: Specified as `integer` in 0.4/latest schemas

File: 0.2/schemas/plate.schema [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+              },
+              "name": {
+                "type": "string"
+              },
+              "description": {
+                "type": "string"
+              },
+              "starttime": {
+                "type": "number"
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:93]: Specified as `integer` in 0.4/latest schemas

File: 0.2/schemas/plate.schema [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+              "maximumfieldcount": {
+                "type": "number"
+              },
+              "name": {
+                "type": "string"
+              },
+              "description": {
+                "type": "string"
+              },
+              "starttime": {
```
Comment (sbesson) [inline 0.2/schemas/plate.schema:92]: Should we also have `endttime`?

File: 0.3/schemas/plate.schema [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+          "description": "Maximum number of fields per view across all wells."
+        },
+        "acquisitions": {
+          "description": "Rows of the Plate grid",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "id": {
+                "type": "number"
```
Comment (sbesson) [inline 0.3/schemas/plate.schema:81]: Same as above

File: 0.3/schemas/plate.schema [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+              },
+              "maximumfieldcount": {
+                "type": "number"
```
Comment (sbesson) [inline 0.3/schemas/plate.schema:84]: Same as above

File: 0.3/schemas/plate.schema [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+              },
+              "name": {
+                "type": "string"
+              },
+              "description": {
+                "type": "string"
+              },
+              "starttime": {
+                "type": "number"
```
Comment (sbesson) [inline 0.3/schemas/plate.schema:93]: Same as above

**Review by sbesson — CHANGES_REQUESTED** [review]
A few comments based on manual comparison with  https://github.com/ome/ngff/blob/main/0.4/schemas/plate.schema and https://github.com/ome/ngff/blob/main/0.4/schemas/well.schema.

Ideally, it would be useful to have a JSON test suite for each version/specification with minimally a compliant example so that we can test the validation logic works.

**Comment by will-moore** [comment]
@sbesson Thanks for the fixes etc. I think the other issues are fixed in [09ab485](https://github.com/ome/ngff/pull/127/commits/09ab485d2869fbae8e3346f0694b399613e79b9e)

**Comment by will-moore** [comment]
I'll add tests...

**Comment by will-moore** [comment]
"feels like an overhead for the scope of this work" -> you think it should be done in this PR, or it outside the scope of this PR?
Pushed fix to 0.1 plate.schema.

**Review by sbesson — APPROVED** [review]

**Comment by sbesson** [comment]
> you think it should be done in this PR, or it outside the scope of this PR?

Not worth it imo unless we start adding more content to these old versions schemas.

**Comment by will-moore** [comment]
I'm missing something. Having merged this, I'd expect https://github.com/ome/ngff/tree/main/0.3/schemas to show plate.schema but I don't see it there??

**Comment by sbesson** [comment]
@joshmoore would this be related to the afternoon force push events?

**Pull request merged**